[Documentation] HF MCP Server (#1938)
huggingface/hub-docs: docs/hub/hf-mcp-server.md

- Run community tools via MCP‑compatible Gradio apps hosted on [Spaces](https://hf.co/spaces).
- Bring results back into your assistant with metadata, links, and context.

## Get started

1. Open your MCP settings: visit https://huggingface.co/settings/mcp while logged in.

2. Pick your client: select your MCP‑compatible client (for example Cursor, VS Code, Zed, Claude Desktop). The page shows client‑specific instructions and a ready‑to‑copy configuration snippet.

3. Paste and restart: copy the snippet into your client’s MCP configuration, save, and restart/reload the client. You should see “Hugging Face” (or similar) listed as a connected MCP server in your client.

<Tip>

The settings page generates the exact configuration your client expects. Use it rather than writing config by hand.

</Tip>

![MCP Settings Example](https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/hf-mcp-settings.png)

## Using the server

After connecting, ask your assistant to use the Hugging Face tools. Example prompts:

- “Search Hugging Face models for Qwen 3 Quantizatizations.”
- “Find a Space that can transcribe audio files.”
- “Show datasets about weather time‑series.”

> - “Create a 1024 x 1024 image of a cat anime style.”

Author: ```suggestion
- “Create a 1024 x 1024 image of a cat ghibli style.”
```

# Hugging Face MCP Server

> The Hugging Face MCP (Model Context Protocol) Server connects your MCP‑compatible AI assistant (for example Codex, Cursor, VS Code extensions, Zed, or Claude Desktop) directly to the Hugging Face Hub. Once connected, your assistant can search and explore Hub resources and use community tools, all from within your editor.

Reviewer: ```suggestion
The Hugging Face MCP (Model Context Protocol) Server connects your MCP‑compatible AI assistant (for example Codex, Cursor, VS Code extensions, Zed, or Claude Desktop) directly to the Hugging Face Hub. Once connected, your assistant can search and explore Hub resources and use community tools, all from within your editor or chat.
```

## What you can do

- Search and explore Hub resources: models, datasets, Spaces, and papers
- Run community tools via MCP‑compatible Gradio apps hosted on [Spaces](https://hf.co/spaces)
- Bring results back into your assistant with metadata, links, and context

## Get started

1. Open your MCP settings: visit https://huggingface.co/settings/mcp while logged in.

2. Pick your client: select your MCP‑compatible client (for example Cursor, VS Code, Zed, Claude Desktop). The page shows client‑specific instructions and a ready‑to‑copy configuration snippet.

3. Paste and restart: copy the snippet into your client’s MCP configuration, save, and restart/reload the client. You should see “Hugging Face” (or similar) listed as a connected MCP server in your client.

<Tip>

> The settings page generates the exact configuration your client expects. Use it rather than writing config by hand.

Reviewer: ```suggestion

The settings page generates the exact configuration your client expects. Use it rather than writing config by hand.

```

## Using the server

After connecting, ask your assistant to use the Hugging Face tools. Example prompts:

- “Search Hugging Face models for Qwen 3 Quantizatizations.”
- “Find a Space that can transcribe audio files.”
- “Show datasets about weather time‑series.”
- “Create a 1024 x 1024 image of a dof anime style.”

Your assistant will call MCP tools exposed by the HF MCP Server (including HF spaces) and return results (titles, owners, downloads, links, and so on). You can then open the resource on the Hub or continue iterating in the same chat.

![HF MCP with Spaces in VS Code](https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/hf-mcp-vscode.png)

## Add community tools (Spaces)

You can extend your setup with MCP‑compatible Gradio spaces built by the community:

- Explore Spaces with MCP support [here](https://huggingface.co/spaces?filter=mcp-server).
- Add the relevant space in your MCP settings on Hugging Face [here](https://huggingface.co/settings/mcp).

Gradio MCP apps expose their functions as tools (with arguments and descriptions) so your assistant can call them directly.

> 

Reviewer: You could add some visuals with cool spaces. There some in existing content like the [blog post](https://huggingface.co/blog/gradio-mcp).

```<video src="https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/gradio-guides/mcp_guide1.mp4" style="width:100%" controls preload> </video>```

## Security and permissions

- Scoped access: the server uses your authenticated Hugging Face account to access Hub resources. You can review or revoke access anytime from your Hugging Face settings.
- Least surprise: tools only perform actions you prompt them to do in your assistant; browsing and lookups are read‑only.
- Privacy: results returned to your client are based on your account’s visibility and permissions (for example, private org content remains private).

## Learn more

- Settings and client setup: https://huggingface.co/settings/mcp
- Changelog announcement: https://huggingface.co/changelog/hf-mcp-server
- HF MCP Server: https://huggingface.co/mcp

> 

Reviewer: - Build your own MCP Server with Gradio Spaces: https://www.gradio.app/guides/building-mcp-server-with-gradio

# Hugging Face MCP Server

> The Hugging Face MCP (Model Context Protocol) Server connects your MCP‑compatible AI assistant (for example Codex, Cursor, VS Code extensions, Zed, or Claude Desktop) directly to the Hugging Face Hub. Once connected, your assistant can search and explore Hub resources and use community tools, all from within your editor or chat.

Reviewer 2: I don't see Zed in the settings right now.

Author: you can still use it via `others` and it's a seemingly popular IDE: https://zed.dev/docs/ai/mcp

## What you can do

- Search and explore Hub resources: models, datasets, Spaces, and papers
- Run community tools via MCP‑compatible Gradio apps hosted on [Spaces](https://hf.co/spaces)
- Bring results back into your assistant with metadata, links, and context

## Get started

1. Open your MCP settings: visit https://huggingface.co/settings/mcp while logged in.

2. Pick your client: select your MCP‑compatible client (for example Cursor, VS Code, Zed, Claude Desktop). The page shows client‑specific instructions and a ready‑to‑copy configuration snippet.

> 3. Paste and restart: copy the snippet into your client’s MCP configuration, save, and restart/reload the client. You should see “Hugging Face” (or similar) listed as a connected MCP server in your client.

Reviewer 2: Shouldn't we recommend to choose additional tools before restarting? Or maybe we can mention it later.

<Tip>

The settings page generates the exact configuration your client expects. Use it rather than writing config by hand.

</Tip>

![MCP Settings Example](https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/hf-mcp-settings.png)

## Using the server

After connecting, ask your assistant to use the Hugging Face tools. Example prompts:

- “Search Hugging Face models for Qwen 3 Quantizatizations.”
- “Find a Space that can transcribe audio files.”
- “Show datasets about weather time‑series.”

> - “Create a 1024 x 1024 image of a dof anime style.”

Reviewer 2: ```suggestion
- “Create a 1024 x 1024 image of a dog anime style.”
```

I guess?

Author: it's supposed to be a cat (see the image below :p)

Reviewer 2: Yeah, but the prompt is different, I thought it was a different example. If you want we can sync the prompt in the markdown to `create an image of a cat ghibli style 1024 x 1024`

Author: sure, done here: https://github.com/huggingface/hub-docs/pull/1938/commits/043a9d84fd01e3887b8b9192a32ffac9aa08912f

> Your assistant will call MCP tools exposed by the HF MCP Server (including HF spaces) and return results (titles, owners, downloads, links, and so on). You can then open the resource on the Hub or continue iterating in the same chat.

Reviewer 2: ```suggestion
Your assistant will call MCP tools exposed by the Hugging Face MCP Server (including Spaces) and return results (titles, owners, downloads, links, and so on). You can then open the resource on the Hub or continue iterating in the same chat.
```

![HF MCP with Spaces in VS Code](https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/hf-mcp-vscode.png)

## Add community tools (Spaces)

You can extend your setup with MCP‑compatible Gradio spaces built by the community:

- Explore Spaces with MCP support [here](https://huggingface.co/spaces?filter=mcp-server).
- Add the relevant space in your MCP settings on Hugging Face [here](https://huggingface.co/settings/mcp).

> Gradio MCP apps expose their functions as tools (with arguments and descriptions) so your assistant can call them directly.

Reviewer 2: ```suggestion
Gradio MCP apps expose their functions as tools (with arguments and descriptions) so your assistant can call them directly. Please, restart or refresh your client so it picks up new tools you select.
```

## Security and permissions

- Scoped access: the server uses your authenticated Hugging Face account to access Hub resources. You can review or revoke access anytime from your Hugging Face settings.

> - Least surprise: tools only perform actions you prompt them to do in your assistant; browsing and lookups are read‑only.

Reviewer 2: I'm not sure what this means.

> - Privacy: results returned to your client are based on your account’s visibility and permissions (for example, private org content remains private).

Reviewer 2: But if my token has permissions for private repos it will find them, won't it?

Author: I removed the entire section.

## Learn more

- Settings and client setup: https://huggingface.co/settings/mcp
- Changelog announcement: https://huggingface.co/changelog/hf-mcp-server

> - HF MCP Server: https://huggingface.co/mcp

Reviewer 2: ```suggestion
- Hugging Face MCP Server: https://huggingface.co/mcp
```
